### ISSUE TITLE
Instance rejected prematurely

### ISSUE THREAD

**Issue by fonsecadeline** [VROOM-Project/vroom#98]
Hi, 

When trying to solve the following instance, I get the error message : "Too high cost values, stopping to avoid overflowing".

```
{"vrp":{"points":[
{"id":"depot","location":{"lat":43.698883,"lon":7.18244}},
{"id":11578,"location":{"lat":"43.742798","lon":"7.4241676"}},
{"id":11579,"location":{"lat":"43.740894","lon":"7.4177237"}},
{"id":11744,"location":{"lat":"27.241438","lon":"6.2586102"}}],

"vehicles":[
{"id":"A101","start_point_id":"depot"}],

"services":[
{"id":11578,"type":"service","sticky_vehicle_ids":["A101"],"activity":{"point_id":11578}},
{"id":11579,"type":"service","sticky_vehicle_ids":["A101"],"activity":{"point_id":11579}}],

"configuration":{"resolution":{"duration":30000,"iterations_without_improvment":100,"initial_time_out":20000,"time_out_multiplier":2}}}}
```

Indeed, point with ID = 11744 has an infinite distance defined in our preprocessing. However, there is no service at this point. Therefore, this infinite distance should not be taken into account.

**Comment by jcoupey** [comment]
The check your instance is failing on was introduced to prevent overflowing when "infinite distances" are added in custom matrices (see #50 for more background).

I guess you have a pre-processing step on the above input before feeding it to VROOM, so it would be helpful if you could post this example formatted following [the VROOM API](https://github.com/VROOM-Project/vroom/blob/master/docs/API.md#input).

**Comment by fonsecadeline** [comment]
Here is the instance provided to VROOM : 

> {"vehicles":[{"id":0,"start_index":0,"end_index":0}],"jobs":[{"id":0,"location_index":1},{"id":1,"location_index":2}],"matrix":[[0,2545,2129,2147483647],[2434,0,653,2147483647],[2094,563,0,2147483647],[2147483647,2147483647,2147483647,0]]}

**Comment by jcoupey** [comment]
Thanks. What happens is indeed that the upper bound computation operates on the whole matrix.

We should store the ranks of the matrix that are actually used when defining jobs and vehicles in [`input::add_job`](https://github.com/VROOM-Project/vroom/blob/master/src/structures/vroom/input/input.cpp#L20) and [`input::add_vehicle`](https://github.com/VROOM-Project/vroom/blob/master/src/structures/vroom/input/input.cpp#L47). Then only loop over those when computing the max costs per line/column in [`input::check_cost_bound`](https://github.com/VROOM-Project/vroom/blob/master/src/structures/vroom/input/input.cpp#L125).

Note that you'll get the expected result by removing the unused lines and columns at index 3 in your instance. So in the meantime you could just compute your matrix based on the `services` and `vehicles` start/end in your initial input, rather than on `points`.

**Comment by jcoupey** [comment]
@fonsecadeline feel free to open a PR for this. ;-)

**Comment by fonsecadeline** [comment]
Thank you for your answer ! Yes, I already considered removing lines/columns associated to services removed during preprocessing. Still I thought that would be something interesting for you to know ;)

**Comment by jcoupey** [comment]
Solved with #100. Thanks @fonsecadeline for noticing and fixing.
